src/lib/y2users/config_manager.rb

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "singleton"
+require "y2users/config"
+
+module Y2Users
+  # Holds references to different configuration instances.
+  # It is not mandatory for config to be registered here.
+  class ConfigManager
+    include Singleton
+
+    def initialize
+      @register = {}
+    end
+
+    # Register config as given id
+    # @param [Config] config
+    # @param [Symbol] as id of config
+    # @note if given id is already registered, it is overwritten
+    # rubocop:disable Naming/UncommunicativeMethodParamName
+    def register(config, as:)
+      @register[as] = config
```

Comment:
Just to prevent errors, I would ensure `as` is a symbol here. Either by doing `register[as.to_sym] = ...` or by raising an `ArgumentError` if it's not. Otherwise I fear the situation we have in other parts in which we happily mix strings and symbols as keys.

Reply:
And I would do something similar in `#unregister` and `#config`

Reply:
I would probably use Argument error and discuss if we should allow more in future as in general any object can be used.